File: homework/hw-4/hw4-starter-code/cpp/sql_test_1.cpp

#include <iostream>
#include <pqxx/pqxx>
#include "config.h"

using namespace std;
using namespace pqxx;

int main()
{
  try {
    // connection info
    string usr = config::USER;
    string pwd = config::PASSWORD;
    string hst = config::HOST;
    string dat = config::DATABASE;
    string url = "postgresql://" + usr + ":" + pwd + "@" + hst + "/" + dat;
    
    // create a connection
    connection cx{url};
    work tx{cx};

    // execute a query
    string q = "SELECT id, name FROM pet ORDER BY name";
    result r{tx.exec(q)};
    for (auto row: r) {
      cout << row["id"].as<int>() << ", " << row["name"].c_str() << endl;
    }

    // not needed here, but good habit if updates
    tx.commit();
  }
  catch (sql_error const &e) {
    cerr << "SQL error: " << e.what() << endl;
    return 1;
  }
  catch (exception const &e) {
    cerr << "Error: " << e.what() << endl;
    return 1;
  }
}

File: homework/hw-4/hw4-starter-code/java/SQLTest1.java

import java.io.FileInputStream;
import java.io.IOException;
import java.sql.Connection;
import java.sql.DriverManager;
import java.sql.ResultSet;
import java.sql.SQLException;
import java.sql.Statement;
import java.util.Properties;

public class SQLTest1 {

  public static void main(String[] args) {
    try {
      // connection info
      String url = "jdbc:postgresql://cps-postgresql.gonzaga.edu/";
      Properties props = new Properties();
      FileInputStream in = new FileInputStream("config.properties");
      props.load(in);
      in.close();
      
      // connect to the DBMS
      Connection cn = DriverManager.getConnection(url, props);

      // create a (non-prepared) statement
      Statement st = cn.createStatement();

      // execute a simple, non-parameterized query 
      String q = "SELECT id, name FROM pet ORDER BY name";
      ResultSet rs = st.executeQuery(q);
      
      // print results
      while(rs.next()) {
        int id = rs.getInt("id");
        String name = rs.getString("name");
        System.out.println(id + ", " + name);
      }

      // release resources
      rs.close();
      st.close();
      cn.close();
    }
    catch(Exception e) {
      e.printStackTrace();
    }
  }
  
}
